freqtrade/data/converter/orderflow.py

```diff
@@ -0,0 +1,277 @@
+                        imbalance_volume=config_orderflow["imbalance_volume"],
+                    )
+                )
+
+                _stacked_imb = config_orderflow["stacked_imbalance_range"]
+                df.loc[is_between, "stacked_imbalances_bid"] = df.loc[
+                    is_between, "imbalances"
+                ].apply(lambda x: stacked_imbalance_bid(x, stacked_imbalance_range=_stacked_imb))
+                df.loc[is_between, "stacked_imbalances_ask"] = df.loc[
+                    is_between, "imbalances"
+                ].apply(lambda x: stacked_imbalance_ask(x, stacked_imbalance_range=_stacked_imb))
+
+                buy = df.loc[is_between, "bid"].apply(
+                    lambda _: np.where(
+                        trades_grouped_df["side"].str.contains("buy"),
+                        0,
+                        trades_grouped_df["amount"],
+                    )
+                )
+                sell = df.loc[is_between, "ask"].apply(
+                    lambda _: np.where(
+                        trades_grouped_df["side"].str.contains("sell"),
+                        0,
+                        trades_grouped_df["amount"],
+                    )
+                )
+                deltas_per_trade = sell - buy
+                min_delta = 0
+                max_delta = 0
+                delta = 0
+                for deltas in deltas_per_trade:
+                    for d in deltas:
+                        delta += d
+                        if delta > max_delta:
+                            max_delta = delta
+                        if delta < min_delta:
+                            min_delta = delta
+                df.loc[is_between, "max_delta"] = max_delta
+                df.loc[is_between, "min_delta"] = min_delta
+
+                df.loc[is_between, "bid"] = np.where(
+                    trades_grouped_df["side"].str.contains("buy"), 0, trades_grouped_df["amount"]
+                ).sum()
+                df.loc[is_between, "ask"] = np.where(
+                    trades_grouped_df["side"].str.contains("sell"), 0, trades_grouped_df["amount"]
+                ).sum()
+                df.loc[is_between, "delta"] = df.loc[is_between, "ask"] - df.loc[is_between, "bid"]
+                min_delta = np.min(deltas_per_trade)
+                max_delta = np.max(deltas_per_trade)
+
+                df.loc[is_between, "total_trades"] = len(trades_grouped_df)
+                # copy to avoid memory leaks
+                dataframe.loc[is_between] = df.loc[is_between].copy()
+            else:
+                logger.debug(f"Found NO candles for trades starting with {candle_start}")
+        logger.debug(f"trades.groups_keys in {time.time() - start_time} seconds")
+
+        logger.debug(f"trades.singleton_iterate in {time.time() - start_time} seconds")
+
+    except Exception as e:
+        logger.exception("Error populating dataframe with trades:", e)
+        raise e
```

Comment:
i think this should raise a "freqtrade-ish" exception instead? otherwise we catch it here - but still have any random exception bubble "everywhere" ...

Reply:
what's a "freqtrade-ish" exception for example?

Reply:
Depends what we want to accomplish i think
most likely, something around `DependencyException` (which doesn't completely crash the bot) should work.

ValueError can also be used - the point is though - it should be one error - not "all possible errors python has to offer" 😆